subscription-engine/kafka-ws/app/Index.js

```diff
@@ -0,0 +1,62 @@
+
+const { Logger } = require('@dojot/microservice-sdk');
+const { app: { log: logConfig } } = require('./Config');
+const Server = require('./Server');
+
+
+Logger.setTransport('console', {
+  level: logConfig.log_console_level,
+});
+
+if (logConfig.log_file) {
+  Logger.setTransport('file', {
+    level: logConfig.log_file_level,
+    filename: logConfig.log_file_filename,
+  });
+}
+
+Logger.setVerbose(logConfig.log_verbose);
+
+const logger = new Logger();
+
+const unhandledRejections = new Map();
+// the unhandledRejections Map will grow and shrink over time,
+// reflecting rejections that start unhandled and then become handled.
+process.on('unhandledRejection', (reason, promise) => {
+  // The 'unhandledRejection' event is emitted whenever a Promise is rejected and
+  // no error handler is attached to the promise within a turn of the event loop.
+  logger.error(`Unhandled Rejection at: ${reason.stack || reason}.`);
+  unhandledRejections.set(promise, reason);
+  logger.debug(`unhandledRejection: List of Unhandled Rejection size ${unhandledRejections.size}`);
+});
+process.on('rejectionHandled', (promise) => {
+  // The 'rejectionHandled' event is emitted whenever a Promise has
+  // been rejected and an error handler was attached to it
+  // later than one turn of the Node.js event loop.
+  logger.debug('rejectionHandled: A event');
+  unhandledRejections.delete(promise);
+  logger.debug(`rejectionHandled: List of Unhandled Rejection size ${unhandledRejections.size}`);
+});
+
+process.on('uncaughtException', async (ex) => {
+  // The 'uncaughtException' event is emitted when an uncaught JavaScript
+  // exception bubbles all the way back to the event loop.
+  logger.error(`uncaughtException: Unhandled Exception at: ${ex.stack || ex}. Bailing out!!`);
+  // TODO: stop server (connection redis, kafka consumer, etc.)
+  process.exit(1);
```

Comment:
As the [NodeJS docs](https://nodejs.org/docs/latest-v10.x/api/process.html#process_process_exit_code) says, the correct way to exit the program is by doing `process.exitCode = 1` and not calling `process.exit(1)`.

Reply:
I'm not sure if a graceful shutdown is the best choice here. Even when you are doing a graceful shutdown, it is necessary a timely call to force a shutdown if the graceful stop doesn't happen in a predefined time interval.

But, this a good issue to be discussed latter and applies to all our services.